feat: cp-7.47.0 solana code fences

aganglada opened the pull request:
## **Description**

- enables the Solana and Non-EVM code by moving the code fence flags to the main set.
- this change makes all the Solana and Non-EVM features available in main build.
- updates e2e tests to take the new modal into account

> [!NOTE]
> The `keyring-snaps` is the code fence flag that allows for snaps like the `solana`, `bitcoin`, any keyring snap to work, the Solana snap and bitcoin snap are keyring-snaps.
> This code fence has all the broad/general code for non-evm snaps, and then for specific network logic we have the solana, bitcoin  etc...

## **Related issues**

Fixes: the current feature is under code fencing, only available in our beta build, it has to be moved from beta to main set to be available in the next release.

## **Manual testing steps**

```gherkin
Feature: Solana

  Scenario: user opens new app install
    Given the app is installed and wallet created
    When user lands on wallet screen
    Then user sees the new Solana feature announcement bottom sheet
    And user sees in the accounts list "add Solana account"
    And user sees in the network tab "Solana"

  Scenario: user opens updated app
    Given the app is updated with wallet already created
    When user logs in
    Then user sees the new Solana feature announcement bottom sheet
    And user sees in the accounts list "add Solana account"
    And user sees in the network tab "Solana"
```

## **Screenshots/Recordings**

### **Before**

#### Add account
<img width="340" alt="add account screenshot" src="https://github.com/user-attachments/assets/2473cd0b-cfcd-4c4a-8f8e-73143e16315d" />

#### Select network
<img width="340" alt="select network screenshot" src="https://github.com/user-attachments/assets/632f52cc-1013-4573-853d-178e372e220c" />


### **After**

#### New welcome bottom sheet

<img width="340" alt="new welcome modal screenshot" src="https://github.com/user-attachments/assets/8124a0cd-e4b2-4daf-bdef-c08e7ff18516" />


#### Add account
<img width="340" alt="add account screenshot" src="https://github.com/user-attachments/assets/5771e51a-6688-4d26-91df-7d7da622feba" />


#### Select network
<img width="340" alt="select network screenshot" src="https://github.com/user-attachments/assets/54e3a06c-1d65-41dc-a2c2-eef8fb5a3c37" />

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

aganglada committed:
feat: solana code fences

zone-live committed:
Merge branch 'main' into feat/solana-code-fences

zone-live committed:
Merge branch 'main' into feat/solana-code-fences

NicolasMassart reviewed (dismissed):
I tested (built the feature, provided screenshots in description) and I approve the changes, we just have to make sure the e2e failures are not related to this PR to merge.
QA team can help on this. Contact them please.

Andepande committed:
test: resolve failing smoke tests caused Solana integration (#15382)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

Got it! Here's an updated PR description that includes those details:

---

Fixes existing smoke E2E test failures that occur when the Solana
feature is enabled.

* Suppresses the Solana feature sheet for tests that use fixtures.
* For tests that don’t use fixtures, automatically swipes down to
dismiss the modal.

This ensures stability across smoke tests regardless of the Solana
feature flag state.

---

Let me know if you'd like to mention specific test files or tag
reviewers.

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes:


https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/1a3fd4be-4f11-42b7-a2a7-a6c8b14f399a

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

NicolasMassart committed:
Merge branch 'main' into feat/solana-code-fences

NicolasMassart reviewed (approved):
I approve these e2e changes and addition of test ids in the UI
